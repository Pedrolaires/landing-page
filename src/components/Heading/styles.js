import styled, { css } from "styled-components";

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  large: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (upperCase) => css`
  text-transform: ${upperCase ? "uppercase" : "none"};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, upperCase }) => css`
    color: ${colorDark
      ? theme.colors.lightGrayColor
      : theme.colors.primaryColor};

    ${titleSize[size](theme)};
    ${titleCase(upperCase)};
  `}
`;
