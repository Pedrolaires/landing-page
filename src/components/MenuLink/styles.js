import styled, { css } from "styled-components";
import theme from "../../styles/theme";

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;
    background: blue;

    &::after {
      content: "";
      position: absolute;
      bottom: 0;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
