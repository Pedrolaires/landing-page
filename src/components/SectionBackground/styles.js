import styled, { css } from "styled-components";

const containerBackgroundActivate = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.whiteColor};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.whiteColor};
    color: ${theme.colors.primaryColor};
    ${background && containerBackgroundActivate(theme)};

    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
