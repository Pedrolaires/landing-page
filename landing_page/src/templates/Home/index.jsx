import { useEffect, useRef, useState } from "react";

import * as Styled from "./styles";
import { mapData } from "../../api/map-data";

import { GridTwoColumns } from "../../components/GridTwoColumns";
import { GridContent } from "../../components/GridContent";
import { Heading } from "../../components/Heading";
import { Base } from "../Base";
import { PageNotFound } from "../PageNotFound";
import { Loading } from "../Loading";
import { GridText } from "../../components/GridText";
import { GridImage } from "../../components/GridImage";

function Home() {
  const [data, setData] = useState([]);
  const isMounted = useRef(true);

  useEffect(() => {
    const load = async () => {
      try {
        const data = await fetch(
          "http://localhost:1337/api/pages/?slug=landing-page",
        );
        const json = await data.json();
        const pageData = mapData(json);
        setData(pageData[0]);
      } catch (e) {
        setData(undefined);
      }
    };
    if (isMounted) {
      load();
    }
    return () => {
      isMounted.current = false;
    };
  }, []);

  if (data === undefined) {
    return <PageNotFound />;
  }

  if (data && !data.slug) {
    return <Loading />;
  }

  const { menu, sections, footerHtml, slug } = data;
  const { links, text, link, imageSrc } = menu;
  console.log(sections);
  return (
    <Base
      links={links}
      footerHtml={footerHtml}
      logoData={{ text, link, imageSrc }}
    >
      {sections.map((section, index) => {
        const { component, __component } = section;
        //console.log(__component);
        //http://localhost:1337/api/pages?populate=deep
        const key = `${slug}-${index}`;
        if (__component) {
          if (__component === "section.section-grid-text") {
            return <GridText key={key} {...section} />;
          }
          if (__component === "section.section-grid-image") {
            return <GridImage key={key} {...section} />;
          }
        } else {
          if (component === "section.section-two-columns") {
            return <GridTwoColumns key={key} {...section} />;
          }
          if (component === "section.section-content") {
            return <GridContent key={key} {...section} />;
          }
        }
      })}
    </Base>
  );
}

export default Home;
