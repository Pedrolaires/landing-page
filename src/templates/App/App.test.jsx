import { screen } from "@testing-library/react";
import Home from ".";
import { renderTheme } from "../../styles/render-theme";
import { theme } from "../../styles/theme";

test("render <Home /> component.", () => {
  renderTheme(<Home />);
});

test("Background do h1 da match com a cor segundaria do tema", () => {
  renderTheme(<Home />);
  const headingContainer = screen.getByRole("heading", {
    name: "Oi",
  }).parentElement;

  expect(headingContainer).toHaveStyleRule(
    "background",
    theme.colors.secondaryBg,
  );
  expect(headingContainer).toMatchSnapshot();
});
