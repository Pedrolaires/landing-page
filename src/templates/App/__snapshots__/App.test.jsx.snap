// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background do h1 da match com a cor segundaria do tema 1`] = `
.c0 {
  background: #55D;
}

<div
  class="c0"
>
  <h1>
    Oi
  </h1>
</div>
`;
